experimental/FTheoryTools/src/G4Fluxes/special_attributes.jl

```diff
@@ -638,3 +640,347 @@ function well_quantized_and_vertical_ambient_space_models_of_g4_fluxes(m::Abstra
+
+julia> res = well_quantized_and_vertical_and_no_non_abelian_gauge_group_breaking_ambient_space_models_of_g4_fluxes(t, check = false);
+
+julia> res[1]
+2 by 0 empty matrix
+
+julia> res[2]
+2 by 0 empty matrix
+```
+
+Here is a more interesting example, in which we verify with our software tool for one particular F-theory QSM, that the
+choice of $G_4$-flux presented in [CHLLT19](@cite), is indeed vertical, satisfies the necessary conditions
+for being well-quantized and does not break the non-Abelian gauge group.
+
+```jldoctest; setup = :(Oscar.LazyArtifacts.ensure_artifact_installed("QSMDB", Oscar.LazyArtifacts.find_artifacts_toml(Oscar.oscardir)))
+julia> qsm_model = literature_model(arxiv_id = "1903.00009", model_parameters = Dict("k" => 2021))
+Hypersurface model over a concrete base
+
+julia> res = well_quantized_and_vertical_and_no_non_abelian_gauge_group_breaking_ambient_space_models_of_g4_fluxes(qsm_model, check = false);
+
+julia> size(res[1])
+(37, 1)
+
+julia> size(res[2])
+(37, 0)
+
+julia> M=res[1];
+
+julia> g4_base = ambient_space_models_of_g4_fluxes(qsm_model, check = false);
+
+julia> g4_classes = [sum(M[i,j]*g4_base[i] for i in 1:length(g4_base)) for j in 1:size(M,2)];
+
+julia> length(g4_classes) == 1
+true
+
+julia> g4_classes[end]
+Cohomology class on a normal toric variety given by 293//300*x4*e2 + 143//150*x4*u - 283//25*x4*e4 + 143//150*x4*e1 + 1643//300*x4*w - 599//150*x5*x8 - 7//150*x5*e2 - 7//75*x5*u - 7//50*x5*e4 - 7//75*x5*e1 - 89//300*x5*w + 896//75*x6*x7 + 1//20*x6*e2 + 1//10*x6*u + 2//5*x6*e4 + 1//10*x6*e1 - 1//5*x6*w - 599//75*x7*x8 - 7//75*x7*e2 - 14//75*x7*u - 7//25*x7*e4 - 14//75*x7*e1 - 89//150*x7*w + 208//75*x8^2 + 298//75*x8*x9 + 1//150*x8*e2 - 73//150*x8*u - 12//25*x8*e4 - 73//150*x8*e1 - 37//75*x8*w + 82//75*x9^2 - 7//150*x9*e2 - 7//75*x9*u + 9//25*x9*e4 - 7//75*x9*e1 - 41//75*x9*w + 11//30*e1*w
+
+julia> g4_candidate = g4_flux(qsm_model, g4_classes[end], check = false);
+
+julia> passes_elementary_quantization_checks(g4_candidate)
+true
+
+julia> passes_verticality_checks(g4_candidate)
+true
+
+julia> breaks_non_abelian_gauge_group(g4_candidate)
+false
+
+julia> c = [60, 51, 90, 0, 24, 51, -24, 45, 30, 0, -48, 90, -57, 60, 30, 15, 120, 0, -60, 0, -720, -420, -270, -60, -2190];
+
+julia> qsm_g4_candidate = g4_flux(qsm_model, sum(c[i]*g4_classes[i] for i in 1:length(g4_classes)), check = false)
+G4-flux candidate
+  - Elementary quantization checks: not executed
+  - Tadpole cancellation check: not executed
+  - Verticality checks: not executed
+  - Non-Abelian gauge group: breaking pattern not analyzed
+
+julia> passes_elementary_quantization_checks(qsm_g4_candidate)
+true
+
+julia> passes_verticality_checks(qsm_g4_candidate)
+true
+
+julia> breaks_non_abelian_gauge_group(qsm_g4_candidate)
+false
+```
+"""
+function well_quantized_and_vertical_and_no_non_abelian_gauge_group_breaking_ambient_space_models_of_g4_fluxes(m::AbstractFTheoryModel; check::Bool = true)::Tuple{QQMatrix, QQMatrix}
```

Comment:
```suggestion
function well_quantized_and_vertical_and_no_non_abelian_gauge_group_breaking_ambient_space_models_of_g4_fluxes(m::AbstractFTheoryModel; check::Bool = true)
```
adding the type here introduces usually unwanted calls to `convert` during lowering. Please see my other comments for the appropriate place to but the return type